src/main/java/com/nvidia/spark/rapids/jni/ParseURI.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.jni;
+
+import ai.rapids.cudf.ColumnVector;
+import ai.rapids.cudf.ColumnView;
+import ai.rapids.cudf.DType;
+import ai.rapids.cudf.NativeDepsLoader;
+
+public class ParseURI {
+  static {
+    NativeDepsLoader.loadNativeDeps();
+  }
+
+
+  /**
+   * Parse protocol for each URI from the incoming column.
+   *
+   * @param URIColumn The input strings column in which each row contains a URI.
+   * @return A string column with protocol data extracted.
+   */
+  public static ColumnVector parseURIProtocol(ColumnView URIColumn) {
+    assert URIColumn.getType().equals(DType.STRING) : "Input type must be String";
+    return new ColumnVector(parseProtocol(URIColumn.getNativeView()));
```

Comment:
Nit: Variable names start with a lower case letter, otherwise they look like type names.
```suggestion
  public static ColumnVector parseURIProtocol(ColumnView uriColumn) {
    assert uriColumn.getType().equals(DType.STRING) : "Input type must be String";
    return new ColumnVector(parseProtocol(uriColumn.getNativeView()));
```